{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "8d51a887",
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "from pathlib import Path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/lus/flare/projects/FoundEpidem/siebenschuh/aurora_gpt/small-pdf-dataset/01_of_20.pdf\n",
      "/lus/flare/projects/FoundEpidem/siebenschuh/aurora_gpt/small-pdf-dataset/16_of_20.pdf\n",
      "/lus/flare/projects/FoundEpidem/siebenschuh/aurora_gpt/small-pdf-dataset/06_of_20.pdf\n",
      "/lus/flare/projects/FoundEpidem/siebenschuh/aurora_gpt/small-pdf-dataset/19_of_20.pdf\n",
      "/lus/flare/projects/FoundEpidem/siebenschuh/aurora_gpt/small-pdf-dataset/02_of_20.pdf\n",
      "/lus/flare/projects/FoundEpidem/siebenschuh/aurora_gpt/small-pdf-dataset/15_of_20.pdf\n",
      "/lus/flare/projects/FoundEpidem/siebenschuh/aurora_gpt/small-pdf-dataset/17_of_20.pdf\n",
      "/lus/flare/projects/FoundEpidem/siebenschuh/aurora_gpt/small-pdf-dataset/18_of_20.pdf\n",
      "/lus/flare/projects/FoundEpidem/siebenschuh/aurora_gpt/small-pdf-dataset/03_of_20.pdf\n",
      "/lus/flare/projects/FoundEpidem/siebenschuh/aurora_gpt/small-pdf-dataset/07_of_20.pdf\n",
      "/lus/flare/projects/FoundEpidem/siebenschuh/aurora_gpt/small-pdf-dataset/04_of_20.pdf\n",
      "/lus/flare/projects/FoundEpidem/siebenschuh/aurora_gpt/small-pdf-dataset/13_of_20.pdf\n",
      "/lus/flare/projects/FoundEpidem/siebenschuh/aurora_gpt/small-pdf-dataset/11_of_20.pdf\n",
      "/lus/flare/projects/FoundEpidem/siebenschuh/aurora_gpt/small-pdf-dataset/20_of_20.pdf\n",
      "/lus/flare/projects/FoundEpidem/siebenschuh/aurora_gpt/small-pdf-dataset/10_of_20.pdf\n"
     ]
    }
   ],
   "source": [
    "dir_path = Path('/lus/flare/projects/FoundEpidem/siebenschuh/aurora_gpt/output/nougat/new_nougat/parsed_pdfs')\n",
    "\n",
    "# loop jsonl\n",
    "good_pages  = 0\n",
    "fail_pages  = 0\n",
    "empty_pages = 0\n",
    "\n",
    "fail_dict = {}\n",
    "empty_dict = {}\n",
    "\n",
    "# 1st: loop jsonl\n",
    "for jsonl_file in dir_path.glob('*.jsonl'):\n",
    "    # loop docs\n",
    "    with open(jsonl_file, 'r') as f:\n",
    "        for j,line in enumerate(f):\n",
    "            dd = json.loads(line)\n",
    "\n",
    "            # doc text\n",
    "            idx = dd['metadata']['page_char_idx']\n",
    "            text = dd['text']\n",
    "            path = dd['path']\n",
    "\n",
    "            print(path)\n",
    "\n",
    "            # loop pages\n",
    "            for page_no in range(len(idx)-1):\n",
    "                page_text = text[idx[page_no] : idx[page_no+1]]\n",
    "                if 'MISSING_PAGE_FAIL' in page_text:\n",
    "                    fail_pages+=1\n",
    "                    # TODO: append page_no to list fail_dict['path'].append()\n",
    "                    fail_dict.setdefault(path, []).append(page_no)\n",
    "                elif 'MISSING_PAGE_EMPTY' in page_text:\n",
    "                    empty_pages+=1\n",
    "                    # TODO: append page_no to list empty_dict['path'].append()\n",
    "                    empty_dict.setdefault(path, []).append(page_no)\n",
    "                else:\n",
    "                    good_pages+=1\n",
    "\n",
    "        #print(path)\n",
    "        #print(path)\n",
    "        #if path=='/lus/flare/projects/FoundEpidem/siebenschuh/aurora_gpt/small-pdf-dataset/01_of_20.pdf':\n",
    "        #    print(jsonl_file)\n",
    "        #    #break\n",
    "        #if path=='/lus/flare/projects/FoundEpidem/siebenschuh/aurora_gpt/output/nougat/new_nougat/parsed_pdfs/c8d23375-e7bd-4533-b839-2f686120797e.jsonl':\n",
    "        #    break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "4f58ff28",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(0, 0, 299)"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fail_pages, empty_pages, good_pages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "2d80f1ac",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'2 \\nbetween treatment, anatomical, and patient-related variables. These interactions can \\nsignificantly influence treatment efficacy and patient prognosis [4]. Additionally, the use of \\nradiotherapy varies significantly across different cancer diagnoses, and understanding these \\nvariations can help improve treatment strategies [5]. GBM and LGG are particularly interesting \\nto study together because GBM often originates from a preexisting LGG, representing a \\nprogression from a lower-grade to a higher-grade malignancy [6]. This progression is \\nassociated with significant changes in gene expression profiles [7], which may underlie the \\ndifferences in radiotherapy outcomes observed between these two cancer types. \\nUnderstanding these differences is crucial, as it can inform personalized treatment strategies \\nand improve survival outcomes. \\n \\nRecent advancements in radiotherapy for GBM and LGG include the exploration of various \\ntargeted therapies, novel radiotherapy approaches, and immunotherapies. For instance, \\nvaccine strategies have shown promising results in early-phase clinical studies [8]. Particle \\nirradiation and dose escalation strategies, including modern molecular imaging, are being \\nevaluated for their long-term outcomes [9]. Intraoperative radiotherapy (IORT) is being \\nexplored to sterilize the margins from persistent tumor cells and bridge the therapeutic gap \\nbetween surgery and radiochemotherapy [10]. Immunotherapy is another promising modality, \\nwith radiotherapy potentially enhancing the effect of immunotherapy through various \\nmechanisms [11]. \\n \\nDespite these advancements, the survival rate for GBM patients has not significantly improved, \\nindicating a need for novel anti-target therapies that could be used in conjunction with standard \\nradiochemotherapy approaches [12]. Radiotherapy resistance is frequently observed in GBM \\npatients and is a major cause of the high mortality rate [13]. This resistance is often \\nmultifactorial and heterogeneous, associated with the recurrence of GBM after surgery [14]. \\nFor LGG, while radiotherapy is effective, it is associated with neurological, cognitive, and \\nendocrinological morbidity, prompting the use of chemotherapy regimens aimed at delaying \\nradiotherapy, especially in younger children [15].  \\n \\nIn response to these challenges, our study aims to investigate the variability in radiotherapy \\noutcomes between GBM and LGG patients. We collected radiotherapy phenotype data for 32 \\nTCGA cancer datasets and performed an overall survival analysis for 32 cancer types. We \\nalso conducted a signaling pathway enrichment analysis to uncover the underlying biological \\nprocesses contributing to the observed differences in radiotherapy outcomes for GBM and \\nLGG cancer patients. The differential regulation of pathways related to programmed cell death, \\nDNA repair, telomere maintenance, chromosome condensation, antiviral responses, and \\ninterferon signaling was observed. The findings could have a significant impact on \\npersonalized treatment approaches and novel co-treatment approaches with RT.  \\n \\nMaterials and Methods \\n \\nData Collection and Differential Expression Analysis \\nGene expression and patient phenotype data for 32 TCGA cancers were collected from the \\nUCSC Xena database [16]. FPKM-UQ expression data obtained from tumor samples were \\nuploaded into PandOmics [17] and pre-processed according to the PandaOmics pipeline. \\nDifferential expression analysis was performed using the limma package for TCGA-GBM and \\nTCGA-LGG datasets, comparing patients who received radiotherapy to those who did not \\nAll rights reserved. No reuse allowed without permission. \\n(which was not certified by peer review) is the author/funder, who has granted medRxiv a license to display the preprint in perpetuity. \\nThe copyright holder for this preprint\\nthis version posted July 5, 2024. \\n; \\nhttps://doi.org/10.1101/2024.07.04.24309952\\ndoi: \\nmedRxiv preprint \\nbetween treatment, anatomical, and patient-related variables. These interactions can significantly influence treatment efficacy and patient prognosis [4]. Additionally, the use of radiotherapy varies significantly across different cancer diagnoses, and understanding these variations can help improve treatment strategies [5]. GBM and LGG are particularly interesting to study together because GBM often originates from a preexisting LGG, representing a progression from a lower-grade to a higher-grade malignancy [6]. This progression is associated with significant changes in gene expression profiles [7], which may underlie the differences in radiotherapy outcomes observed between these two cancer types. Understanding these differences is crucial, as it can inform personalized treatment strategies and improve survival outcomes.\\n\\nRecent advancements in radiotherapy for GBM and LGG include the exploration of various targeted therapies, novel radiotherapy approaches, and immunotherapies. For instance, vaccine strategies have shown promising results in early-phase clinical studies [8]. Particle irradiation and dose escalation strategies, including modern molecular imaging, are being evaluated for their long-term outcomes [9]. Intraoperative radiotherapy (IORT) is being explored to sterilize the margins from persistent tumor cells and bridge the therapeutic gap between surgery and radiochemotherapy [10]. Immunotherapy is another promising modality, with radiotherapy potentially enhancing the effect of immunotherapy through various mechanisms [11].\\n\\nDespite these advancements, the survival rate for GBM patients has not significantly improved, indicating a need for novel anti-target therapies that could be used in conjunction with standard radiochemotherapy approaches [12]. Radiotherapy resistance is frequently observed in GBM patients and is a major cause of the high mortality rate [13]. This resistance is often multifactorial and heterogeneous, associated with the recurrence of GBM after surgery [14]. For LGG, while radiotherapy is effective, it is associated with neurological, cognitive, and endocrinological morbidity, prompting the use of chemotherapy regimens aimed at delaying radiotherapy, especially in younger children [15].\\n\\nIn response to these challenges, our study aims to investigate the variability in radiotherapy outcomes between GBM and LGG patients. We collected radiotherapy phenotype data for 32 TCGA cancer datasets and performed an overall survival analysis for 32 cancer types. We also conducted a signaling pathway enrichment analysis to uncover the underlying biological processes contributing to the observed differences in radiotherapy outcomes for GBM and LGG cancer patients. The differential regulation of pathways related to programmed cell death, DNA repair, telomere maintenance, chromosome condensation, antiviral responses, and interferon signaling was observed. The findings could have a significant impact on personalized treatment approaches and novel co-treatment approaches with RT.\\n\\n## Materials and Methods\\n\\n### Data Collection and Differential Expression Analysis\\n\\nGene expression and patient phenotype data for 32 TCGA cancers were collected from the UCSC Xena database [16]. FPKM-UQ expression data obtained from tumor samples were uploaded into PandOmics [17] and pre-processed according to the PandaOmics pipeline. Differential expression analysis was performed using the limma package for TCGA-GBM and TCGA-LGG datasets, comparing patients who received radiotherapy to those who did not receive radiotherapy. Each dataset was processed according to standard protocols. The obtained gene-wise p-values were corrected using the Benjamini-Hochberg procedure. Differential expression results were later used for gene set enrichment analysis.\\n\\n#### Overall survival analysis of IR-treated and IR-untreated cancer patients\\n\\nSurvival analysis was conducted in PandaOmics using the KaplanMeierFitter function from the lifelines Python package. Patients were divided into two groups: those who received radiotherapy and those who did not receive radiotherapy. Only patients with available expression data were included in the analysis. Briefly, 32 TCGA cancers were analyzed, and survival analysis was performed between the two described groups of patients. The log-rank test was used to calculate statistical significance. The significance of survival outcomes was plotted on a heatmap and colored red if radiotherapy increased survival outcomes and blue if the application of radiotherapy decreased survival outcomes. Non-significant results were colored white. Combined survival plots for patients who received and did not receive radiotherapy across all TCGA cancers were plotted using the matplotlib package.\\n\\n#### Signaling pathway enrichment analysis\\n\\nPathway enrichment analysis was performed using the gseapy package with the enrich() function, following standard protocols. The Reactome database was selected as the source of gene sets from the gseapy internal library for signaling pathway enrichment analysis. Genes that were significantly up-regulated in TCGA-GBM and simultaneously down-regulated in TCGA-LGG, and vice versa, were used as input for the pathway enrichment analysis. The top 20 significantly perturbed signaling pathways were visualized on a dot plot using the gseapy.plot function.\\n\\n#### Paper draft preparation\\n\\nThe draft of this paper was generated using DORA, Insilico Medicine\\'s LLM-based paper drafting assistant. Draft Outline Research Assistant (DORA) is designed to streamline the process of publication creation, making it faster and simpler. The process of paper generation is curated by over 30 AI agents, powered by Large Language Models (LLMs), and integrated with internal and other curated databases, to assist in generating high-quality scientific papers. Each agent employs Retrieval-Augmented Generation (RAG) to perform comprehensive data collection and analysis, reduce the probability of hallucinations, and provide relevant PubMed links to make the generation of the paper more transparent. Followed by generation, the draft was manually curated and extended by the authors.\\n\\nmedRxiv preprint doi: [https://doi.org/10.1101/2024.07.04.24309595](https://doi.org/10.1101/2024.07.04.24309595); this version posted July 5, 2024. The copyright holder for this preprint (which was not certified by peer review) is the author/fundedr.w. has granted andRxiv a license to display the preprint in perpetuity. All rights reserved. No reuse allowed without permission.\\n\\n## Results\\n\\n### Patient and Tumor Characteristics in TCGA Cancers\\n\\nRadiotherapy phenotype data were collected for 32 TCGA cancer datasets from the UCSC Xena database. TCGA-LAML was excluded from the analysis since there were no patients who received radiotherapy. The total number of samples varies between 1,194 and 45 for TCGA-BRCA and TCGA-CHOL, respectively. Similarly, the percentage of patients who received radiotherapy varies between 82% and 0% for TCGA-GBM and TCGA-KIRC/TCGA-CHOL/TCGA-KICH/TCGA-KIRP, respectively. It was noted that TCGA-GBM and TCGA-LGG are the cancers with the highest percentage of patients who received radiotherapy, at 82% and 54%, respectively (Figure 1).\\n\\n### Overall survival analysis of patients who received and did not receive radiotherapy\\n\\nTo study whether radiotherapy treatment could be used as a trait capable of stratifying cancer patients with different outcomes, we performed an overall survival analysis for 32 cancer types from the TCGA dataset. The overall survival analysis was conducted for patients who received radiotherapy and for those who did not receive radiotherapy (Figure 2, A). In some cases, radiotherapy can improve survival outcomes, while in others, it worsens them. For example, it was found that patients with GBM, BRCA, READ, UCEC, STAD, and HNSC who received radiotherapy lived longer compared to patients who did not receive radiotherapy. Conversely, it was noted that patients with UVM, LUAD, and LGG who received radiotherapy had shorter survival compared to those who did not receive radiotherapy (Figure 2, B). This observation led us to focus on the differences between GBM and LGG patients who received and did not receive radiotherapy, since GBM can develop from LGG.\\n\\nFigure 1: **Overview of the patient’s samples downloaded from TCGA. Number of samples for each cancer type is shown, sorted by the total number of samples. Stacked bars are colored according to the sample category, including non-irradiated control samples, irradiated control samples, irradiated case samples, non-irradiated case samples, control samples without information about radiotherapy and case samples without information about radiotherapy.**\\n\\nmedRxiv preprint doi: [https://doi.org/10.1101/022047.04.24309595](https://doi.org/10.1101/022047.04.24309595); this version posted July 5, 2024. The copyright holder for this preprint (which was not certified by peer review) is the author/author, who has granted andRxiv a license to display the preprint in perpetuity.\\n\\n_Signaling pathway enrichment analysis of GBM and LGG patients_\\n\\nIn order to find differences between GBM and LGG patients who were exposed to IR and who weren\\'t on the transcriptomic level, we collected gene expression data for those patients. Then we created comparisons between patients who received radiotherapy and those who did not, and calculated differentially expressed genes. After that, we obtained a list of genes that were significantly down-regulated in GBM patients and at the same time significantly up-regulated in LGG patients, and vice versa. Those gene lists were used for pathway enrichment analysis to uncover the underlying biological processes contributing to the observed differences in radiotherapy outcomes (Figure 3). It was noted that pathways related to programmed cell death and DNA repair, such as \"Diseases Of Programmed Cell Death\" and \"Base-Excision Repair,\" were down-regulated in GBM but up-regulated in LGG. This suggests a potential mechanism by which LGG cells might be more susceptible to radiotherapy-induced damage, whereas GBM cells might evade such damage. It was also observed that pathways involved in telomere maintenance and chromosome condensation, such as \"Packaging Of Telomere Ends\" and \"Condensation Of Prophase Chromosomes,\" were differentially regulated, indicating a possible role in the differential radiotherapy outcomes between GBM and LGG (Figure 3, A). Moreover, pathways associated with antiviral responses and interferon signaling, including \"ISG15 Antiviral Mechanism\" and \"Interferon Signaling,\" were up-regulated in GBM and down-regulated in LGG. This could imply an enhanced immune response in GBM, potentially contributing to its resistance to radiotherapy. Finally, pathways related to gene\\n\\nFigure 2: **Survival analysis across 22 TCGA cancers. A, Survival curves for patients who received radiotherapy and those who didn’t presented on a Kaplan-Meier plot. B, The significance of survival results is plotted on a heatmap and colored red if radiotherapy increased survival outcomes and blue if the application of radiotherapy decreased survival outcomes. Non-significant results were colored white (p-value > 0.05). The log-rank test was used to calculate statistical significance.**\\n\\nexpression and protein metabolism, such as \"Gene Expression (Transcription)\" and \"Metabolism Of Proteins,\" were up-regulated in GBM and down-regulated in LGG, suggesting a higher metabolic and transcriptional activity in GBM that might support its aggressive nature and resistance to treatment (Figure 3, B).\\n\\n## Discussion\\n\\nThe study presented here shows the variability in radiotherapy outcomes across different cancer types, with a specific focus on glioblastoma multiforme (GBM) and low-grade gliomas (LGG). One of the unique aspects of this study is its focus on the differential radiotherapy outcomes between GBM and LGG. While radiotherapy remains a fundamental treatment for GBM, the high resistance to treatment and subsequent poor prognosis emphasize the need for novel therapeutic strategies [18]. Conversely, LGG patients often show a more favorable response to radiotherapy, which can be predicted by MRI evaluations post-treatment [19]. The findings from this study have the potential to impact clinical practices and treatment protocols by providing a more detailed understanding of how radiotherapy should be tailored to individual patients based on their specific cancer type.\\n\\nRadiotherapy was found to improve survival outcomes in GBM while worsening outcomes in LGG. GBM and LGG, despite their differences, share a common lineage, with GBM often developing from LGG [6]. This study addresses a critical gap in the existing literature by providing a detailed comparison of radiotherapy outcomes in these two cancer types. The differential regulation of pathways related to programmed cell death, DNA repair, telomere maintenance, chromosome condensation, antiviral responses, and interferon signaling between IR-exposed and not non-IR-exposed GBM and LGG patients may explain the underlying reasons for these observed differences. For instance, the down-regulation of DNA repair pathways in GBM suggests a mechanism for radiotherapy resistance, while their upregulation in LGG indicates a higher susceptibility to radiotherapy-induced damage. The use of radiotherapy in combination with temozolomide has been shown to improve survival rates\\n\\nFigure 3: Signaling pathway enrichment analysis results. A, Signaling pathways enriched with genes significantly down-regulated in GBM patients and at the same time significantly up-regulated in LGG patients. B, Signaling pathways enriched with genes significantly up-regulated in GBM patients and at the same time significantly down-regulated in LGG patients.\\n\\nin GBM patients [20]. However, radiotherapy resistance remains a significant challenge, often leading to poor outcomes [13]. In LGG, the timing and dosage of radiotherapy are crucial factors that can influence survival outcomes [21]. Given the up-regulation of DNA repair pathways in LGG, combining radiotherapy with DNA repair inhibitors could make LGG cells more vulnerable to treatment. For example, using PARP inhibitors could enhance the effectiveness of radiotherapy [22]. On the other hand, the down-regulation of antiviral and interferon signaling pathways in LGG indicates a less active immune environment. Combining radiotherapy with immune modulators, such as interferon therapy or immune checkpoint inhibitors, could boost the immune response against LGG cells and improve treatment outcomes [23]. The results of this study align with these findings, further emphasizing the importance of personalized treatment approaches.\\n\\nIn conclusion, this study provides valuable insights into the variability in radiotherapy outcomes across different cancer types, with a specific focus on GBM and LGG. The identification of key pathways involved in radiotherapy resistance and sensitivity offers potential targets for future therapeutic strategies. The findings highlight the importance of personalized treatment approaches and the need for further research to improve radiotherapy outcomes in cancer patients.\\n\\n## References\\n\\n* [1] Wen P, Gao Y, Chen B, Qi X, Hu G, Xu A, et al. Pan-Cancer Analysis of Radiotherapy Benefits and Immune Infiltration in Multiple Human Cancers. Cancers. 2020;12. doi:10.3390/cancers12040957\\n* [2] Yan D, Zhao Q, Du Z, Li H, Geng R, Yang W, et al. Development and validation of an immune-related gene signature for predicting the radiosensitivity of lower-grade gliomas. Sci Rep. 2022;12. 6698. doi:10.1038/s41598-022-10601-5\\n* [3] Meehan J, Gray M, Martinez-Perez C, Kay C, Pang LY, Fraser JA, et al. Precision Medicine and the Role of Biomarkers of Radiotherapy Response in Breast Cancer. Front Oncol. 2020;10. 628. doi:10.3389/fonc.2020.00628\\n* [4] El Naqa I, Bradley JD, Lindsay PE, Hope AJ, Deasy JO. Predicting radiotherapy outcomes using statistical learning techniques. Phys Med Biol. 2009;54: S9-S30. doi:10.1088/0031-9155/54/18/S02\\n* [5] Walker B, Kavadi V, Wilfong L, Robert N. Practice radiation patterns among oncologists in the Oncology Care Model. Am J Manag Care. 2022;28: 515-519. doi:10.37765/ajmc.2022.89249\\n* [6] Bready D, Placantonakis DG. Molecular Pathogenesis of Low-Grade Glioma. Neurosurg Clin N Am. 2019;30: 17-25. doi:10.1016/j.nec.2018.08.011\\n* [7] Xu Y, Geng R, Yuan F \\'en, Sun Q, Liu B, Chen Q. Identification of differentially expressed key genes between glioblastoma and low-grade glioma by bioinformatics analysis. PeerJ. 2019;7: e6560. doi:10.7717/peerj.6560\\n* [8] Liu EK, Sulman EP, Wen PY, Kurz SC. Novel Therapies for Glioblastoma. Curr Neurol Neurosci Rep. 2020;20: 19. doi:10.1007/s11910-020-01042-6\\n* [9] Combs SE, Schmid TE, Vaupel P, Muthoff G. Stress Response Leading to Resistance in Glioblastoma-The Need for Innovative Radiotherapy (iRT) Concepts.\\n\\nCancers. 2016;8. doi:10.3390/cancers8010015\\n* [10] Giordano FA, Brehmer S, Abo-Madyan Y, Welzel G, Sperk E, Keller A, et al. INTRAGO: intraoperative radiotherapy in glioblastoma multiforme--a phase I/II dose escalation study. BMC Cancer. 2014;14: 992. doi:10.1186/1471-2407-14-992\\n* [11] El-Khayat SM, Arafat WO. Therapeutic strategies of recurrent glioblastoma and its molecular pathways \"Lock up the beast.\" Ecancermedicalscience. 2021;15: 1176. doi:10.3332/ecancer.2021.1176\\n* [12] Mattei V, Santilli F, Martellucci S, Delle Monache S, Fabrizi J, Colapietro A, et al. The Importance of Tumor Stem Cells in Glioblastoma Resistance to Therapy. Int J Mol Sci. 2021;22. doi:10.3390/ijms22083863\\n* [13] Li R, Wang H, Liang Q, Chen L, Ren J. Radiotherapy for glioblastoma: clinical issues and nanotechnology strategies. Biomater Sci. 2022;10: 892-908. doi:10.1039/d1bm01401c\\n* [14] Zhao Y, Kang J-H, Yoo K-C, Kang S-G, Lee H-J, Lee S-J. K-RAS Acts as a Critical Regulator of CD44 to Promote the Invasiveness and Stemness of GBM in Response to Ionizing Radiation. Int J Mol Sci. 2021;22. doi:10.3390/ijms222010923\\n* [15] Lober RM. Low grade gliomas of childhood: the actual management. Minerva Pediatr. 2013;65: 149-165. Available: [https://www.ncbi.nlm.nih.gov/pubmed/23612260](https://www.ncbi.nlm.nih.gov/pubmed/23612260)\\n* [16] Goldman MJ, Craft B, Hastie M, Repecka K, McDade F, Kamath A, et al. Visualizing and interpreting cancer genomics data via the Xena platform. Nat Biotechnol. 2020;38: 675-678. doi:10.1038/s41587-020-0546-8\\n* [17] Kamya P, Ozerov IV, Pun FW, Tretina K, Fokina T, Chen S, et al. PandaOmics: An Al-Driven Platform for Therapeutic Target and Biomarker Discovery. J Chem Inf Model. 2024;64: 3961-3969. doi:10.1021/acs.jcim.3c01619\\n* [18] Corso CD, Bindra RS, Mehta MP. The role of radiation in treating glioblastoma: here to stay. J Neurooncol. 2017;134: 479-485. doi:10.1007/s11060-016-2348-x\\n* [19] Spych M, Gottwald L, Jesien-Lewandowicz E, Sztajer S, Fijuth J. Response to postoperative radiotherapy as a prognostic factor for patients with low-grade gliomas. Oncol Lett. 2012;4: 455-460. doi:10.3892/ol.2012.759\\n* [20] Jalali R, Basu A, Gupta T, Munshi A, Menon H, Sarin R, et al. Encouraging experience of concomitant Temozolomide with radiotherapy followed by adjuvant Temozolomide in newly diagnosed glioblastoma multiforme: single institution experience. Br J Neurosurg. 2007;21: 583-587. doi:10.1080/02688690701604574\\n* [21] Prabhu RS, Ward MC, Heinzerling JH, Corso CD, Buchwald ZS, Dhakal R, et al. The Association Between Radiation Therapy Dose and Overall Survival in Patients With Intracranial Infiltrative Low-Grade Glioma Treated With Concurrent and/or Adjuvant Chemotherapy. Adv Radiat Oncol. 2021;6: 100577. doi:10.1016/j.adro.2020.09.017\\n* [22] Sun C, Chu A, Song R, Liu S, Chai T, Wang X, et al. PARP inhibitors combined with radiotherapy: are we ready? Front Pharmacol. 2023;14: 1234973. doi:10.3389/fphar.2023.123497323. Goedegebuure RSA, Vonk C, Kooij LP, Derks S, Thijssen VLJL. Combining Radiation Therapy With Interferons: Back to the Future. Int J Radiat Oncol Biol Phys. 2020;108: 56-69. doi:10.1016/j.ijrobp.2020.02.016\\n\\n## Authors contribution\\n\\nAV conducted and coordinated in silico studies, and wrote the manuscript; GVM, IVO contributed to the writing of the manuscript; ANO and AA provided helpful suggestions to the experimental design and improvements to the project; AZ and MSK conceived the project; MSK supervised the project, contributed to the writing of the manuscript and is the corresponding author.\\n\\n## Funding sources\\n\\nThis work has been funded by the Danish Cancer society (#R368-A21521, #R302-A17379_001 and #R167-A11015_001), the Novo Nordisk Foundation Challenge Programme (NNF17OC0027812), the Nordea Foundation (02-2017-1749), the Lundbeck Foundation (R324-2019-1492), the Danish Ministry of Higher Education and Science (0238-00003B) and Insilico Medicine. The funders had no role in the study design, data collection and analysis, decision to publish, or manuscript preparation.\\n\\n## Conflicts of interests\\n\\nInsilico Medicine is a company developing an AI-based end-to-end integrated pipeline for drug discovery and development and engaged in aging and cancer research. AV, IVO, AA, and AZ are affiliated with Insilico Medicine.\\n\\n## Data availability\\n\\nAll data supporting the conclusions of the paper are available in the article and corresponding figures. TCGA datasets used in the paper are described in the materials and methods section.\\n\\n## Figure legends\\n\\n## Figure 1. Overview of the patient\\'s samples downloaded from TCGA.\\n\\nNumber of samples for each cancer type is shown, sorted by the total number of samples. Stacked bars are colored according to the sample category, including non-irradiated control samples, irradiated control samples, irradiated case samples, non-irradiated case samples, control samples without information about radiotherapy and case samples without information about radiotherapy.\\n\\n## Figure 2. Survival analysis across 22 TCGA cancers.\\n\\nA, Survival curves for patients who received radiotherapy and those who didn\\'t presented on a Kaplan-Meier plot. B, The significance of survival results is plotted on a heatmap and colored red if radiotherapy increased survival outcomes and blue if the application of radiotherapy decreased survival outcomes. Non-significant results were colored white (p-value > 0.05). The log-rank test was used to calculate statistical significance.\\n\\nFigure 2. Survival analysis across 22 TCGA cancers.\\n\\nA, Survival curves for patients who received radiotherapy and those who didn’t presented on a Kaplan-Meier plot. B, The significance of survival results is plotted on a heatmap and colored red if radiotherapy increased survival outcomes and blue if the application of radiotherapy decreased survival outcomes. Non-significant results were colored white (p-value > 0.05). The log-rank test was used to calculate statistical significance.\\n\\nmedRxiv preprint doi: [https://doi.org/10.1101/02047.04.243095952](https://doi.org/10.1101/02047.04.243095952); this version posted July 5, 2024. The copyright holder for this preprint (which was not certified by peer review) is the author/fundedr.w. has granted andRxiv a license to display the preprint in perpetuity.\\n\\nFigure 3: **Signaling pathway enrichment analysis results.** A, Signaling pathways enriched with genes significantly down-regulated in GBM patients and at the same time significantly up-regulated in LGG patients. B, Signaling pathways enriched with genes significantly up-regulated in GBM patients and at the same time significantly down-regulated in LGG patients.'"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dd['text']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "4091897d",
   "metadata": {},
   "outputs": [],
   "source": [
    "indices = dd['metadata']['page_char_idx']\n",
    "text = dd['text']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "a57b8be7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "medRxiv preprint doi: [https://doi.org/10.1101/2024.07.04.24309595](https://doi.org/10.1101/2024.07.04.24309595); this version posted July 5, 2024. The copyright holder for this preprint (which was not certified by peer review) is the author/author, who has granted and/orku is alone to display the preprint in perpetuity.\n",
      "\n",
      "###### Abstract\n",
      "\n",
      "Radiotherapy is a crucial treatment option for various cancers. However, the results of radiotherapy can vary widely across different cancer types and even among patients with the same type of cancer. This variability presents a major challenge in optimizing treatment strategies and improving patient survival. Here, we collected radiotherapy phenotype and expression data from 32 TCGA cancer datasets and performed overall survival analysis for 32 cancer types. Additionally, we conducted a signalling pathway enrichment analysis to identify key pathways involved in radiotherapy resistance and sensitivity. Our findings show that radiotherapy improves survival outcomes in certain cancer types, such as GBM, while worsening outcomes in others, such as LGG. Next, we focused on exploring the differences in radiotherapy outcomes between GBM and LGG, focusing on the molecular mechanisms contributing to these variations. The differential regulation of pathways related to programmed cell death, DNA repair, telomere maintenance, chromosome condensation, antiviral responses, and interferon signaling between GBM and LGG patients may elucidate the underlying reasons for these observed differences. These insights underscore the importance of personalized treatment approaches and the need for further research to improve radiotherapy outcomes in cancer patients.\n",
      "\n",
      "**Keywords** cancer, GBM, LGG, survival, radiotherapy\n",
      "\n",
      "**Variability in Radiotherapy Outcomes Across Cancer Types: A Comparative Study of Glioblastoma Multiforme and Low-Grade Gliomas**\n",
      "\n",
      "Alexander Veviorskiy\\({}^{1}\\), Garik V Mkrtchyan\\({}^{2}\\), Andreyan N Osipov\\({}^{3}\\), Ivan V Ozerov\\({}^{4}\\), Alex Aliper\\({}^{1}\\), Alex Zhavoronkov\\({}^{1,4,5,6}\\), Morten Scheibye-Knudsen\\({}^{2^{*}}\\)\n",
      "\n",
      "\\({}^{1}\\)Insilico Medicine Al Limited, Abu Dhabi, UAE\n",
      "\n",
      "\\({}^{2}\\)Center for Healthy Aging, Department of Cellular and Molecular Medicine, University of Copenhagen, Denmark\n",
      "\n",
      "\\({}^{3}\\)CANDLE Synchrotron Research Institute, 31 Acharyan, 0040 Yerevan, Armenia\n",
      "\n",
      "\\({}^{4}\\)Insilico Medicine Hong Kong Limited, Hong Kong Science and Technology Park, Hong Kong\n",
      "\n",
      "\\({}^{5}\\)Insilico Medicine Canada Inc., 3710-1250 Rene-Levesque Blvd. W, Montreal, Quebec, Canada H3B 4W8.\n",
      "\n",
      "\\({}^{6}\\)Buck Institute for Research on Aging, Novato, California 94945, United States.\n",
      "\n",
      "\\({}^{*}\\)Corresponding author: mscheibye@sund.ku.dk\n",
      "\n",
      "## Introduction\n",
      "\n",
      "Radiotherapy plays a fundamental role in the treatment of cancer. Even though radiotherapy is widely used, its outcomes can vary significantly depending on the cancer type. It was observed that GBM patients who received \n"
     ]
    }
   ],
   "source": [
    "k = 0\n",
    "i0, i1 = indices[k], indices[k+1]\n",
    "\n",
    "\n",
    "print(text[i0:i1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "3792d348",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/lus/flare/projects/FoundEpidem/siebenschuh/aurora_gpt/small-pdf-dataset/05_of_20.pdf'"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dd['path']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "99a5f747",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!scp /lus/flare/projects/FoundEpidem/siebenschuh/aurora_gpt/small-pdf-dataset/05_of_20.pdf 05_of_20.pdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "65bde8df",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TODO: PYMUPDF PAGE TEXT\n"
     ]
    }
   ],
   "source": [
    "k = 25\n",
    "i0, i1 = indices[k], indices[k+1]\n",
    "\n",
    "\n",
    "print(text[i0:i1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6004df4b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "adaparse",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
